src/explorers/ethereum/etherscan.js

```diff
@@ -0,0 +1,95 @@
+import { BLOCKCHAINS, CONFIG, SUB_STEPS, TRANSACTION_APIS } from '../../constants';
+import { TransactionData, VerifierError } from '../../models';
+import { getText } from '../../domain/i18n/useCases';
+import { stripHashPrefix } from '../utils/stripHashPrefix';
+import { BLOCK_NUMBER_PLACEHOLDER } from '../../constants/api';
+import { request } from '../../services';
+import { buildTransactionApiUrl } from '../../services/transaction-apis';
+
+export function generateTransactionDataFromEtherscanResponse (jsonResponse, isTestChain) {
+  try {
+    // Parse block to get timestamp first, then create TransactionData
+    return getEtherScanBlock(jsonResponse, isTestChain)
+      .then(function (blockResponse) {
+        return parseEtherScanResponse(jsonResponse, blockResponse);
+      })
+      .catch(function () {
+        throw new VerifierError(SUB_STEPS.fetchRemoteHash, getText('errors', 'unableToGetRemoteHash'));
```

Comment:
isn't that the same as what's happening here: https://github.com/blockchain-certificates/cert-verifier-js/pull/393/files#diff-3254a48cbb46ed012658d615c0a96adaR43?

Anyhow we should probably abstract this throw in a specific method that we reuse here and there and in the next `catch`